src/harness/request_handler.py

```diff
@@ -127,12 +131,23 @@ def _Request(url, request, config, is_post_method):
     logging.info('POST Request to URL %s :\n%s', url, request)
   else:
     logging.info('GET Request to URL %s', url)
-  try:
-    conn.perform()
-  except pycurl.error as e:
-    # e contains a tuple (libcurl_error_code, string_description).
-    # See https://curl.haxx.se/libcurl/c/libcurl-errors.html
-    raise CurlError(e.args[1], e.args[0])
+
+  for attempt_count in range(MAX_REQUEST_ATTEMPT_COUNT):
+    try:
+      conn.perform()
+      error_message = ''
+      break
+    except pycurl.error as e:
+      # e contains a tuple (libcurl_error_code, string_description).
+      # See https://curl.haxx.se/libcurl/c/libcurl-errors.html
+      error_message += ' %s\n' % str(CurlError(e.args[1], e.args[0]))
+      time.sleep(REQUEST_ATTEMPT_DELAY_SECOND)
+    except Exception as e:
```

Comment:
Don't you think that it is good to log the error(as info or warning) even if the connection works the next try?
I think that may help for troubleshooting the connectivity problems.

Reply:
+1 This would be very helpful as it would allow ITS to keep track of how often this is happening and, if needed, to investigate alternative/additional remedies for this issue.